zap/src/main/java/org/parosproxy/paros/extension/history/LogPanel.java

```diff
@@ -253,6 +259,35 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         return filterButton;
     }
 
+    private JButton getClearButton() {
+        if (clearButton == null) {
+            clearButton = new JButton();
+            clearButton.setIcon(
+                    new ImageIcon(
+                            LogPanel.class.getResource(
+                                    "/resource/icon/fugue/broom-alert.png")));
+            clearButton.setToolTipText(
+                    Constant.messages.getString("history.clear.tooltip"));
+            DisplayUtils.scaleIcon(clearButton);
+
+            clearButton.addActionListener(
+                    new java.awt.event.ActionListener() {
+
+                        @Override
+                        public void actionPerformed(java.awt.event.ActionEvent e) {
+                            if (View.getSingleton()
+                                    .showConfirmDialog(
+                                            Constant.messages
+                                                    .getString("history.clear.confirm")) == JOptionPane.OK_OPTION) {
+                                extension.getHistoryReferencesTable().selectAll();
+                                extension.purgeHistory(extension.getSelectedHistoryReferences());
+                            }
+                        }
+                    });
```

Comment:
I know you followed the existing examples, but this could probably be simplified with a lambda.

The whole inner if and functionality could be moved to `clear()` or `doClear()` (I didn't check if there's already a clear() method). Then just `clearButton.addActionListener(actionEvent -> clear());`